x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -80,22 +88,43 @@ public TimeValue getTimeout() {
             return timeout;
         }
 
+        public AllocationState getWaitForState() {
+            return waitForState;
+        }
+
+        public Request setWaitForState(AllocationState waitForState) {
+            this.waitForState = ExceptionsHelper.requireNonNull(waitForState, WAIT_FOR);
+            return this;
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
             out.writeString(modelId);
             out.writeTimeValue(timeout);
+            out.writeEnum(waitForState);
         }
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             builder.field(MODEL_ID.getPreferredName(), modelId);
             builder.field(TIMEOUT.getPreferredName(), timeout.getStringRep());
+            builder.field(WAIT_FOR.getPreferredName(), waitForState);
             return builder;
         }
 
         @Override
         public ActionRequestValidationException validate() {
+            if (waitForState.isAnyOf(VALID_WAIT_STATES) == false) {
```

Comment:
`if (waitForState.isAnyOf(VALID_WAIT_STATES)) {
     return null;
}`
`            
 ActionRequestValidationException validationException = new ActionRequestValidationException();
                validationException.addValidationError(
                    "invalid [wait_for] state ["
                        + waitForState
                        + "]; must be one of ["
                        + Strings.arrayToCommaDelimitedString(VALID_WAIT_STATES)
                );
                return validationException;`